src/BenchmarkDotNet/Running/BenchmarkRunnerDirty.cs

```diff
@@ -125,5 +140,59 @@ private static Summary[] RunWithExceptionHandling(Func<Summary[]> run)
                 return new[] { Summary.NothingToRun(e.Message, string.Empty, string.Empty) };
             }
         }
+
+        private static Summary RunWithDirtyAssemblyResolveHelper(IEnumerable<Type> types, IEnumerable<Assembly> assemblies, IConfig config, string[] args)
+        {
+            var logger = config.GetNonNullCompositeLogger();
+            var userInteraction = new UserInteraction();
+            var (isParsingSuccess, parsedConfig, options) = ConfigParser.Parse(args, logger, config);
+            if (!isParsingSuccess) // invalid console args, the ConfigParser printed the error
+                return null;
+
+            if (options.PrintInformation)
+            {
+                logger.WriteLine(HostEnvironmentInfo.GetInformation());
+                return null;
+            }
+
+            var effectiveConfig = ManualConfig.Union(config, parsedConfig);
+            var (allTypesValid, allAvailableTypesWithRunnableBenchmarks) = TypeFilter.GetTypesWithRunnableBenchmarks(types, assemblies, logger);
+            
+            if (!allTypesValid) // there were some invalid and TypeFilter printed errors
+                return null;
+
+            if (allAvailableTypesWithRunnableBenchmarks.IsEmpty())
+            {
+                userInteraction.PrintNoBenchmarksError(logger);
+                return null;
+            }
+
+            if (options.ListBenchmarkCaseMode != ListBenchmarkCaseMode.Disabled)
+            {
+                BenchmarkCasesPrinter.PrintList(logger, effectiveConfig, allAvailableTypesWithRunnableBenchmarks, options);
+                return null;
+            }
+
+            var BenchmarkRunInfos = new List<BenchmarkRunInfo[]>();
+
+            foreach (Type type in types)
+            {
+                var BenchmarkRunInfo = options.UserProvidedFilters
+                               ? TypeFilter.Filter(effectiveConfig, allAvailableTypesWithRunnableBenchmarks)
+                               : new BenchmarkRunInfo[] { BenchmarkConverter.TypeToBenchmarks(type, config) };
+
+                BenchmarkRunInfos.Add(BenchmarkRunInfo);
+            }
+
+            Summary benchmark = BenchmarkRunnerClean.Run(BenchmarkRunInfos.First()).Single();
+
+            foreach (var BenchmarkRunInfo in BenchmarkRunInfos.Skip(1))
+            {
+                benchmark = BenchmarkRunnerClean.Run(BenchmarkRunInfo).Single();
+            }
+
+            return benchmark;
```

Comment:
this could be implemented as following:

```cs
            var filteredBenchmarks = TypeFilter.Filter(effectiveConfig, allAvailableTypesWithRunnableBenchmarks);

            if (filteredBenchmarks.IsEmpty())
            {
                userInteraction.PrintWrongFilterInfo(allAvailableTypesWithRunnableBenchmarks, logger, options.Filters.ToArray());
                return Array.Empty<Summary>();
            }

            return BenchmarkRunnerClean.Run(filteredBenchmarks);
```

this will also most probably require us to return an `IEnumerable<Summary>`, not a single `Summary`. The methods that provide single type and call this method can use `.Single()` on the result of this method

Reply:
that is in case of  ```UserProvidedFilters```, what if user don't provide any?

we also have another case where we create benchmarkruninfro from types.
``` 
 foreach (Type type in types)
            {
                var BenchmarkRunInfo = options.UserProvidedFilters
                               ? TypeFilter.Filter(effectiveConfig, allAvailableTypesWithRunnableBenchmarks)
                               : new BenchmarkRunInfo[] { BenchmarkConverter.TypeToBenchmarks(type, config) };

                BenchmarkRunInfos.Add(BenchmarkRunInfo);
            }
```